util/zenfs.cc

```diff
@@ -142,6 +145,23 @@ void list_children(ZenFS *zenFS, std::string path) {
   }
 }
 
+void format_path(std::string &path)
+{
+  std::size_t pos = path.find('/', 0);
+  while (pos != std::string::npos) {
+    while (path.compare(pos + 1, 1, "/") == 0) {
+      path.erase(pos + 1, 1);
+    }
+    pos = path.find('/', pos + 1);
+  }
+
+  if (path.front() == '/')
```

Comment:
`path.front()` and `path.back()` would return a dangling reference if the string is empty at this point. Shall we add something like
```
if (path.empty()) {
  path = "/";
}
```

here?

Reply:
It turns out the C++ standard guarantees there would be a `\0` if string is empty. No issue here.